### ISSUE TITLE
Timeouts due to new Uptime checks

### ISSUE THREAD

**Issue by netniV** [Cacti/spine#238]
Older devices may not support the new uptime OID that is being used resulting in loss of data due to errors and timeouts.

# Problem - New upTime OID times out
When polling a device, Cacti looks for some basic system information to make sure that the device is up and running, System Uptime is one of these basic bits of information.  It seems that someone recently introduced a second Uptime OID to be polled which gives extended information when you roll beyond a year or two without rebooting.  When I query the device for that OID, snmpwalk simply returns 0 seconds and appears to output it immediately:

```shell
$ snmpwalk -v 2c -c community host:1161 .1.3.6.1.6.3.10.2.1.3.0 SNMP-FRAMEWORK-MIB::snmpEngineTime.0 = INTEGER: 0 seconds 
```

However, the code seems to be experiencing a timeout error obtaining data from the same OID.  At first glance, I thought maybe this was due to the custom port, but given that the other basic info works, it's clearly using the right port.  

```console
2021/11/02 05:03:59 - SPINE: Poller[1] PID[31953] PT[140091731674880] Device[65] DEBUG: snmp_sess_sync_response(.1.3.6.1.6.3.10.2.1.3.0) [complete] 
2021/11/02 05:03:59 - SPINE: Poller[1] PID[31953] PT[140091731674880] ERROR: Failed to get oid '.1.3.6.1.6.3.10.2.1.3.0' for Device[65] with Status[2]
```

One thing I did spot that was wrong, if one of these basic OIDs fails, it marks the host as unreachable to prevent any further delays. So, it would seem that the weird error that shouldn't be an error was flagging the host as non-responsive and thus preventing any data to be queried.

# Correct fatality (don't finish him!)
Make the new OID be non-fatal to ensure it would successfully failback to the other basic sysUptime OID and not think the host was unreachable.  This is achieved by making `snmp_get` become `snmp_get_base` with an extra parameter of `should_fail`.  Calling snmp_get sets that parameter to true.  Any call with the new uptime OID is replaced with snmp_get_base setting that parameter to false.

# Associated Problem
The above fix, when implemented, introduces a new problem.  If there are a large number of retries (system wide setting of 10 for example!) coupled with say 3 second timeouts, any query for this new uptime OID would result in a 30 second delay.  After 30 seconds, it would at least not consider the host as failed, so proceed to query the original uptime OID which works.  

It will then get the rest of the data as required.  As part of this extended set of data, it then seems to query this new uptime OID for a second time (something that needs to be investigated later as to why we aren't just caching that value!).  

As a result, it would delay for another 30 seconds resulting in 60+ seconds and the polling cycle timing out when using 1 minute polling cycles.

# Workaround
For now, to alleviate this, reduce the retries to a lower amount and set any host to timeout after a shorter period.  This should reduce the amount of wasted time on these two unnecessary timeouts.

**Comment by renatolsxc** [comment]
from what I understand this problem was solved in version 1.2.20, correct?  i am using version 1.2.24 and experiencing the same error mentioned and i was not able to request changing timeouts.  the same happens to me, I configure the poller to use spine and several devices that do not have the uptime oid the spine cannot collect valid oid and generate graphs normally using cmd.php.  can you help me?  thank you very much!

**Comment by WirthmU** [comment]
I am using cacti v1.2.27 and spine v.1.3 and get the following error:

`2024-09-01 17:51:02 - SPINE: Poller[Main Poller] PID[150174] PT[140138095117888] ERROR: Failed to get oid '.1.3.6.1.6.3.10.2.1.3.0' for Device[RMS200] with Response[2]`
